GeosCore/diagnostics_mod.F90

```diff
@@ -405,11 +387,48 @@ SUBROUTINE Zero_Diagnostics_StartofTimestep( Input_Opt, State_Diag, RC )
        ENDIF
     ENDIF
 
-    ! Budget diagnostic adjustment for changes in tropopause height
-    ! (transport budget in trop only)
-    IF ( State_Diag%Archive_BudgetTransportTrop ) THEN
-       State_Diag%BudgetTransportTropHeight = 0.0_f8
-    ENDIF
+    ! Budget Diagnostics
+    IF ( State_Diag%Archive_BudgetTransportFull ) &
+         State_Diag%BudgetTransportFull = 0.0_f8
+    IF ( State_Diag%Archive_BudgetTransportTrop ) &
+         State_Diag%BudgetTransportTrop = 0.0_f8
+    IF ( State_Diag%Archive_BudgetTransportPBL ) &
+         State_Diag%BudgetTransportPBL = 0.0_f8
+
+    IF ( State_Diag%Archive_BudgetMixingFull ) &
+         State_Diag%BudgetMixingFull = 0.0_f8
+    IF ( State_Diag%Archive_BudgetMixingTrop ) &
+         State_Diag%BudgetMixingTrop = 0.0_f8
+    IF ( State_Diag%Archive_BudgetMixingPBL ) &
+         State_Diag%BudgetMixingPBL = 0.0_f8
+
+    IF ( State_Diag%Archive_BudgetConvectionFull ) &
+         State_Diag%BudgetConvectionFull = 0.0_f8
+    IF ( State_Diag%Archive_BudgetConvectionTrop ) &
+         State_Diag%BudgetConvectionTrop = 0.0_f8
+    IF ( State_Diag%Archive_BudgetConvectionPBL ) &
+         State_Diag%BudgetConvectionPBL = 0.0_f8
+
+    IF ( State_Diag%Archive_BudgetChemistryFull ) &
+         State_Diag%BudgetChemistryFull = 0.0_f8
+    IF ( State_Diag%Archive_BudgetChemistryTrop ) &
+         State_Diag%BudgetChemistryTrop = 0.0_f8
+    IF ( State_Diag%Archive_BudgetChemistryPBL ) &
+         State_Diag%BudgetChemistryPBL = 0.0_f8
```

Comment:
Chemistry is not called every timestep but here the array is set to zero every timestep. This is going to have implications for time-averages since the history package includes values for every timestep in its computation.

Reply:
This can be fixed by separating out the components that only get executed during chemistry timesteps. However, we cannot use the time_mod.F90 functions to determine if it is a chemistry timestep or not since those functions do not work in non-classic models (including GCHP which uses ESMF for time). Passing a logical for isChemTime would work.